### PR TITLE
Fix GCC build problem with 288f05f related to SmallVector.

### PR DESCRIPTION
**Pull request by shahidact**
Below is the error message for reference.

/llvm-project/mlir/lib/Dialect/Linalg/IR/LinalgOps.cpp: In static member function 'static llvm::SmallVector<mlir::AffineMap> mlir::linalg::MatmulOp::getDefaultIndexingMaps(mlir::MLIRContext*)':
/llvm-project/mlir/lib/Dialect/Linalg/IR/LinalgOps.cpp:3468:10: error: could not convert 'indexingMaps' from 'SmallVector<[...],3>' to 'SmallVector<[...],6>'
 3468 |   return indexingMaps;
      |          ^~~~~~~~~~~~
      |          |
      |          SmallVector<[...],3>

Here is the link to the failure.
https://lab.llvm.org/buildbot/#/builders/117/builds/3919
...

### CODE REVIEW

**Review by rengolin — APPROVED** [review]
Same as before, thanks!

**Review by banach-space — APPROVED** [review]
LGTM, thanks!

Just a small nit :) Dumping a full log from your shell into a Git commit summary can make it hard to parse and less helpful. Could you make it a bit more concise and selective? For example, IMHO, this trimmed version of your log would be sufficient:
```bash
/llvm-project/mlir/lib/Dialect/Linalg/IR/LinalgOps.cpp:3468:10: error: could not convert 'indexingMaps' from 'SmallVector<[...],3>' to 'SmallVector<[...],6>'
3468 | return indexingMaps;
| ^~~~~~~~~~~~
| |
| SmallVector<[...],3>
```

Including a link to the failing bot and the GCC version where this occurs would also be very helpful for context.

Lastly, if no bot is currently failing, it might indicate a gap in our testing. Do you think we should add coverage for this scenario?

**Comment by shahidact** [comment]
> LGTM, thanks!
> 
> Just a small nit :) Dumping a full log from your shell into a Git commit summary can make it hard to parse and less helpful. Could you make it a bit more concise and selective? For example, IMHO, this trimmed version of your log would be sufficient:
> 
> ```shell
> /llvm-project/mlir/lib/Dialect/Linalg/IR/LinalgOps.cpp:3468:10: error: could not convert 'indexingMaps' from 'SmallVector<[...],3>' to 'SmallVector<[...],6>'
> 3468 | return indexingMaps;
> | ^~~~~~~~~~~~
> | |
> | SmallVector<[...],3>
> ```
> 
> Including a link to the failing bot and the GCC version where this occurs would also be very helpful for context.
> 
> Lastly, if no bot is currently failing, it might indicate a gap in our testing. Do you think we should add coverage for this scenario?

Thanks. Yes the failure came from a bot. I will update its link in description.

**Comment by kparzysz** [comment]
This is a blocking breakage.  I'm merging this, since it's been approved.
